pandas/tests/extension/test_arrow.py

```diff
@@ -333,15 +333,39 @@ def test_from_sequence_of_strings_pa_array(self, data, request):
 
 
 class TestGetitemTests(base.BaseGetitemTests):
-    @pytest.mark.xfail(
-        reason=(
-            "data.dtype.type return pyarrow.DataType "
-            "but this (intentionally) returns "
-            "Python scalars or pd.NA"
-        )
-    )
     def test_getitem_scalar(self, data):
-        super().test_getitem_scalar(data)
+        # In the base class we expect data.dtype.type; but this (intentionally)
```

Comment:
@mroeschke might it make sense to reconsider this?  im not sure what context the current `PyArrow.lib.DataType` is really useful

Reply:
Sure, yeah I am not exactly sure how `ExtensionDtype.type` is used internally, but I think it would make sense if it returns the native python type. Would be nice if there was a nice way to derive that from `numpy_dtype`

Reply:
the main use case that comes to mind is if i want to do e.g. `dtype.type(0)` to get an object of the correct type